### ISSUE TITLE
Update to mustangproject 2.15.1 and implement validation

### ISSUE THREAD

**Issue by marzn** [marrelUnderscore/RechnunglessConverter#1]
Hi @marrelUnderscore ,

I have read your comment on https://github.com/paperless-ngx/paperless-ngx/discussions/6823#discussioncomment-11219318 which states you are waiting for an bug fix in the mustangproject. There was a release last week. Does v2.15.1 contains the bug fix already or is there an other blocker open, before we can use the validation from mustangproject here?

Thank you for your great work so far! I though of implementing exactly the same thing, is there anything I can help you with?

**Comment by marrelUnderscore** [comment]
Hi @marzn (interestingly, you have the same name as me) :smile:
The bug actually has already been fixed in V2.14.1 - I just haven't gotten around to continue working on this because life became extremely busy in the past weeks. But today actually was the last day before my christmas vacation, so I'm hoping I can get around to it again in the coming days. I already have a local WIP version where I changed the HTTP-Response to be Json with the PDF embedded as a base64 String in order to also return some metadata (like validation errors, creation date, etc.) in addition to the pdf.  
I wouldn't call it "great work" however - for that it's still way to wonky and improvised, but I really hope I can make it more well rounded in the coming days.

**Comment by marrelUnderscore** [comment]
However, I also like [this](https://github.com/paperless-ngx/paperless-ngx/discussions/6823#discussioncomment-11518801) proposal from the discussion, since it could be fully done in Python and - other than mustangproject - can not only generate german PDFs, but also english ones (and other translations are easily added). But that might be something to look into for the future - I'm first gonna work on the thing I already have right here, because I would like to have something for my own setup at the beginning of the year and german PDFs are what I need anyways.

**Comment by marzn** [comment]
An implementation embedded into paperless-ngx would be nice. What I understand from [this comment](https://github.com/paperless-ngx/paperless-ngx/pull/8488#discussion_r1885390846) is, the possibility to get an implementation for handling XRechnungen to be merged into paperless-ngx is way higher if it has a small footprint in python / paperless-ngx itself and a separate container to convert the file.

**Comment by marrelUnderscore** [comment]
So, after a long session of reworking things and cleaning things up, I now have revision 0.0.3 ready. I hope I haven't overlooked any errors, but my test files all worked as expected. Also updated the readme (over on the RechunglessParser repo) and added an overview of what's currently implemented and what I still need to work on :smile:

**Comment by marzn** [comment]
@marrelUnderscore thank you for your great work 👍
